pandas/core/arrays/boolean.py

```diff
@@ -698,7 +698,15 @@ def _reduce(self, name: str, skipna: bool = True, **kwargs):
 
         if name in {"sum", "prod", "min", "max"}:
             op = getattr(masked_reductions, name)
-            return op(data, mask, skipna=skipna, **kwargs)
+            result = op(data, mask, skipna=skipna, **kwargs)
+
+            # if we have numeric op that would result in an int, coerce to int if possible
+            if name == "prod" and notna(result):
```

Comment:
why did you need to add this back?

Reply:
I was getting a failure on some builds for \tests\arrays\boolean\test_reduction.py where we're checking that the product has int dtype so it might be needed:
```
        elif op == "prod":
>           assert isinstance(getattr(s, op)(), np.int64)
E           AssertionError: assert False
```

Reply:
Hmm, then maybe the test needs to be edited. In any case, we should investigate why it is failing / what we should be expecting. 

From a quick test, it seems that numpy returns int64:

```
In [6]: np.array([], dtype="bool").prod()                                                                                                                                                                          
Out[6]: 1

In [7]: type(_)                                                                                                                                                                                                    
Out[7]: numpy.int64

In [8]: np.array([True, False], dtype="bool").prod()                                                                                                                                                               
Out[8]: 0

In [9]: type(_)                                                                                                                                                                                                    
Out[9]: numpy.int64
```

So I think we should follow that behaviour here

Reply:
From what I can tell it was something that was only affecting Windows builds: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=32986&view=logs&jobId=077026cf-93c0-54aa-45e0-9996ba75f6f7&j=077026cf-93c0-54aa-45e0-9996ba75f6f7&t=e95cf409-86ae-5b4d-6c5f-79395ef75e8f